okapi-core/src/main/java/org/folio/okapi/util/ModuleCache.java

```diff
@@ -0,0 +1,203 @@
+                               HttpMethod method, String uri) {
+    if (re.getProxyType() != RoutingEntry.ProxyType.REDIRECT) {
+      return;
+    }
+    logger.debug("resolveRedirect begin redirectPath={}", re.getRedirectPath());
+    boolean found = false;
+    final String redirectPath = re.getRedirectPath();
+
+    List<ModuleInstance> lookup = lookup(redirectPath, method, filterMap, false, null);
+    for (ModuleInstance instance : lookup) {
+      RoutingEntry tryre = instance.getRoutingEntry();
+      String redirectUri = re.getRedirectUri(uri);
+      found = true;
+      if (routingEntries.add(tryre)) {
+        ModuleInstance mi = new ModuleInstance(instance.getModuleDescriptor(), tryre, redirectUri,
+            method, false);
+        instances.add(mi);
+        resolveRedirect(instances, tryre, loop + " -> " + redirectPath, routingEntries,
+            method, redirectUri);
+      } else {
+        throw new IllegalArgumentException(messages.getMessage("10100", loop, redirectPath));
+      }
+    }
+    lookup = lookup(redirectPath, method, proxyMap, true, null);
+    for (ModuleInstance instance : lookup) {
+      RoutingEntry tryre = instance.getRoutingEntry();
+      String redirectUri = re.getRedirectUri(uri);
+      found = true;
+      if (routingEntries.add(tryre)) {
+        ModuleInstance mi = new ModuleInstance(instance.getModuleDescriptor(), tryre, redirectUri,
+            method, true);
+        instances.add(mi);
+      }
+    }
+    logger.debug("resolveRedirect end redirectPath={} found={}", re.getRedirectPath(), found);
+    if (!found) {
+      throw new IllegalArgumentException(messages.getMessage("10101", uri, redirectPath));
+    }
+  }
+
+  static List<ModuleInstance> lookup(String uri, HttpMethod method, Map<String,
+      List<ModuleCacheEntry>> map, boolean handler, String id) {
+    List<ModuleInstance> instances = new LinkedList<>();
+    String tryUri = uri;
+    for (int index = 0; index < uri.length(); index++) {
+      if (uri.charAt(index) == '#' || uri.charAt(index) == '?') {
+        tryUri = tryUri.substring(0, index);
+        break;
+      }
+    }
+    while (true) {
+      List<ModuleCacheEntry> candidateInstances = map.get(tryUri);
+      if (candidateInstances != null) {
+        for (ModuleCacheEntry candiate : candidateInstances) {
+          if (candiate.routingEntry.match(uri, method.name())
+              && (id == null || id.equals(candiate.moduleDescriptor.getId()))) {
+            instances.add(new ModuleInstance(candiate.moduleDescriptor,
+                candiate.routingEntry, uri, method, handler));
+            if (handler) {
+              return instances;
+            }
+          }
+        }
+      }
+      int index = tryUri.length() - 1;
+      while (index > 0 && tryUri.charAt(index - 1) != '/') {
+        --index;
+      }
+      if (index <= 0) {
+        break;
+      }
+      tryUri = tryUri.substring(0, index);
```

Comment:
We can use lastIndexOf:
```suggestion
      int index = tryUri.lastIndexOf('/', tryUri.length() - 2);
      if (index < 0) {
        break;
      }
      tryUri = tryUri.substring(0, index + 1);
```

Reply:
Definitely shorter.